src/components/molecules/TokenApproval.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { ReactElement, useEffect, useState } from 'react'
+import Button from '../atoms/Button'
+import { useOcean } from '../../providers/Ocean'
+import { useAsset } from '../../providers/Asset'
+import Loader from '../atoms/Loader'
+import { useWeb3 } from '../../providers/Web3'
+import { useUserPreferences } from '../../providers/UserPreferences'
+import Tooltip from '../atoms/Tooltip'
+
+function ButtonApprove({
+  amount,
+  coin,
+  approveTokens,
+  isLoading
+}: {
+  amount: string
+  coin: string
+  approveTokens: (amount: string) => void
+  isLoading: boolean
+}) {
+  const { infiniteApproval } = useUserPreferences()
+
+  const tooltipSpecific = `Give the smart contract permission to spend your ${coin} which has to be done for each transaction. You can optionally set this to infinite in your user preferences.`
+  const tooltipInfinite = `Give the smart contract permission to spend infinte amounts of your ${coin} so you have to do this only once. You can disable allowing infinite amounts in your user preferences.`
+
+  return infiniteApproval ? (
+    <Button
+      style="primary"
+      size="small"
+      onClick={() => approveTokens(`${2 ** 53 - 1}`)}
+    >
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <>
+          Approve {coin} <Tooltip content={tooltipInfinite} />
+        </>
+      )}
+    </Button>
+  ) : (
+    <Button style="primary" size="small" onClick={() => approveTokens(amount)}>
+      {isLoading ? (
+        <Loader />
+      ) : (
+        <>
+          Approve {amount} {coin}
+          {amount === '1' ? 'S' : ''}
+          <Tooltip content={tooltipSpecific} />
+        </>
+      )}
+    </Button>
+  )
+}
+
+export default function TokenApproval({
+  actionButton,
+  amount,
+  coin
+}: {
+  actionButton: JSX.Element
+  disabled: boolean
+  amount: string
+  coin: string
+}): ReactElement {
+  const { ddo } = useAsset()
+  // const [approveToken, setApproveToken] = useState(false)
```

Comment:
was not sure if this is needed at all after I removed the initial switch from approve button to approve deciding buttons, hence I left it commented out throughout code in here. Can be removed once verified it's indeed not needed anymore